File: src/OpenFleet.Domain/Entities/Vendor.cs
namespace OpenFleet.Domain.Entities;

public class Vendor : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string ContactName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;

    public ICollection<Part> Parts { get; set; } = new List<Part>();
}

File: src/OpenFleet.Web/src/types/vendor.ts
export interface VendorFilterRequest {
  search?: string
}

export interface VendorResponse {
  id: string
  name: string
  contactName: string
  email: string
  phone: string
  address: string
  partCount: number
  hasAssignedParts: boolean
  createdAt: string
  updatedAt: string
}

export interface VendorPartSummary {
  id: string
  name: string
  partNumber: string
  quantityOnHand: number
  isLowStock: boolean
}

export interface VendorDetailResponse extends VendorResponse {
  parts: VendorPartSummary[]
}

export interface CreateVendorRequest {
  name: string
  contactName: string
  email: string
  phone: string
  address: string
}

export interface UpdateVendorRequest {
  name?: string
  contactName?: string
  email?: string
  phone?: string
  address?: string
}
